client/document/support-user.jsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+
+function supportUserFn( { user, token, authorized } ) {
+	const url = window.location.toString();
+
+	if ( url.indexOf( '?' ) > 0 ) {
+		const cleanUrl = url.substring( 0, url.indexOf( '?' ) );
+		window.history.replaceState( {}, document.title, cleanUrl );
+	}
+
+	if ( authorized ) {
+		window.sessionStorage.setItem( 'boot_support_user', window.JSON.stringify( { user, token } ) );
+	}
+
+	window.location.replace( '/' );
+}
+
+function SupportUser( { supportUser, supportToken, authorized = false } ) {
+	return (
+		<html lang="en">
+			<body>
+				{ /* eslint-disable react/no-danger */ }
+				<script
+					dangerouslySetInnerHTML={ {
+						__html: `
+						${ supportUserFn.toString() }
+
+						supportUserFn( {
+							user: ${ supportUser && `"${ encodeURIComponent( supportUser ) }"` },
```

Comment:
I'm not sure encodeURIComponent is sufficient. We'd also need to check if the token uses any values that would be encoded. Maybe we want to escape any single and double quotes here?

For example, does this break with input like `",window.alert('hi')`?

Reply:
Just did a quick test in Chrome which seems to play ok:

```
$ encodeURIComponent(`",window.alert('hi')`)
> "%22%2Cwindow.alert('hi')"
```

This is what the response looks like:

<img width="383" alt="screen shot 2018-02-14 at 19 40 36" src="https://user-images.githubusercontent.com/9202899/36221651-141fe450-11bf-11e8-89e8-0d4436ced3a6.png">

Is there any other function in Calypso generally used to escape user provided input?

> We'd also need to check if the token uses any values that would be encoded.

True. Which characters are used for the token?

Reply:
@vindl do you mind taking a quick look at the related code?

Reply:
I think all that `pug` does is encode html angle brackets (< and > ), quotation marks, and a few other items: [pug_escape](https://github.com/pugjs/pug-runtime/blob/master/index.js#L188)

Reply:
> True. Which characters are used for the token?

> @vindl do you mind taking a quick look at the related code?

@gwwar @flootr the token will only contain lowercase hexits, so using `encodeURIComponent` shouldn't cause any problems there.

Reply:
@flootr  `encodeURIComponent` is fine to escape the token/user values.